router/batchrouter/batchrouter.go

```diff
@@ -1563,29 +1566,25 @@ func (worker *workerT) workerProcess() {
 					brtQueryStat := stats.NewTaggedStat("batch_router.jobsdb_query_time", stats.TimerType, map[string]string{"function": "workerProcess"})
 					brtQueryStat.Start()
 					brt.logger.Debugf("BRT: %s: DB about to read for parameter Filters: %v ", brt.destType, parameterFilters)
-					retryList := brt.jobsDB.GetToRetry(
-						jobsdb.GetQueryParamsT{
-							CustomValFilters:              []string{brt.destType},
-							JobsLimit:                     toQuery,
-							ParameterFilters:              parameterFilters,
-							IgnoreCustomValFiltersInQuery: true,
-							PayloadSizeLimit:              brt.payloadLimit,
-						},
-					)
-					unprocessedList := brt.jobsDB.GetUnprocessed(
-						jobsdb.GetQueryParamsT{
-							CustomValFilters:              []string{brt.destType},
-							JobsLimit:                     toQuery,
-							ParameterFilters:              parameterFilters,
-							IgnoreCustomValFiltersInQuery: true,
-							PayloadSizeLimit:              brt.payloadLimit,
-						},
-					)
+					queryParams := jobsdb.GetQueryParamsT{
+						CustomValFilters:              []string{brt.destType},
+						JobsLimit:                     toQuery,
+						ParameterFilters:              parameterFilters,
+						IgnoreCustomValFiltersInQuery: true,
+						PayloadSizeLimit:              brt.payloadLimit,
+					}
+					toRetry := brt.jobsDB.GetToRetry(queryParams)
+					combinedList = toRetry.Jobs
+					if !toRetry.LimitsReached {
+						queryParams.JobsLimit -= len(toRetry.Jobs)
+						if queryParams.PayloadSizeLimit > 0 {
+							queryParams.PayloadSizeLimit -= toRetry.PayloadSize
```

Comment:
How will jobsdb behave if PayloadSizeLimit turns out -ve?

(or there won't be any such scenario?)

Reply:
This will not happen in practive since the `PayloadSize` of the previous response cannot be greater (or equal) than the `PayloadSizeLimit` without `LimitsReached` being `true`.
Setting `PayloadSizeLimit` to any valye `<= 0` effectively disables the limit

Reply:
👍